### PR TITLE
ggml: Add epsilon as a parameter for group_norm

### PR DESCRIPTION
**Pull request by MollySophia**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

Hi! I'm working on adding RWKV implementation in llama.cpp. The graph needs a custom eps value for group_norm.
This PR adds epsilon as a parameter for group_norm operators.

### CODE REVIEW

**Review by NeoZhangJianyu — APPROVED** [review]

File: ggml/src/ggml.c [hunk 1]
```diff
@@ -5369,6 +5370,7 @@ static struct ggml_tensor * ggml_group_norm_impl(
     struct ggml_tensor * result = inplace ? ggml_view_tensor(ctx, a) : ggml_dup_tensor(ctx, a);
 
     result->op_params[0] = n_groups;
+    *(float*)(result->op_params + 1) = eps;
```
Comment (compilade) [inline ggml/src/ggml.c:5373]:
```suggestion
    ggml_set_op_params_i32(result, 0, n_groups);
    ggml_set_op_params_f32(result, 1, eps);
```

I think this should do the same thing

File: ggml/src/ggml.c
Comment (compilade) [inline ggml/src/ggml.c:1]: `ggml_compute_forward_group_norm_f32` needs to be adapted to this too, or else it won't take the epsilon from the params.

File: ggml/src/ggml.c [hunk 2]
```diff
@@ -5380,15 +5382,17 @@ static struct ggml_tensor * ggml_group_norm_impl(
 struct ggml_tensor * ggml_group_norm(
     struct ggml_context * ctx,
     struct ggml_tensor * a,
-    int n_groups) {
-    return ggml_group_norm_impl(ctx, a, n_groups, false);
+    int n_groups,
+    float eps) {
+    return ggml_group_norm_impl(ctx, a, n_groups, eps, false);
```
Comment (compilade) [inline ggml/src/ggml.c:5387]: (Side question) Is `ggml_group_norm` equivalent to `ggml_norm` with `ggml_reshape` to group rows together beforehand and split them back afterwards?

File: ggml/src/ggml.c
Comment (MollySophia) [reply ggml/src/ggml.c:1]: Yes. I missed that one.
Thanks!

File: ggml/src/ggml.c (hunk 2 above)
Comment (MollySophia) [reply ggml/src/ggml.c:5387]: I guess yes?

File: ggml/src/ggml-cann/aclnn_ops.cpp [hunk 3]
```diff
@@ -464,8 +464,8 @@ void ggml_cann_group_norm(ggml_backend_cann_context& ctx, ggml_tensor* dst) {
     aclTensor* acl_src = ggml_cann_create_tensor(src);
     aclTensor* acl_dst = ggml_cann_create_tensor(dst);
 
-    const float eps = 1e-6f;  // TODO: make this a parameter
     int n_groups = dst->op_params[0];
+    const float eps = *(float*)(dst->op_params + 1);
```
Comment (ggerganov) [inline ggml/src/ggml-cann/aclnn_ops.cpp:468]: Use `memcpy` to respect strict aliasing rules

**Review by ggerganov — APPROVED** [review]

**Comment by MollySophia** [comment]
Hmm... It seems that one of the ci checks is failing
What could be the reason?

**Comment by compilade** [comment]
> Hmm... It seems that one of the ci checks is failing
> What could be the reason?

Seems like a timeout. I've seen that test fail before for the same reason, and usually it passes when retried. The timeout is likely too short when all is run at the same time.

**Comment by ggerganov** [comment]
Yeah, that test started failing recently for some reason - definitely not related to this PR. I tried to figure out where to increase the timeout for slot erasure, but could figure out. Would be nice if fix this eventually

**Pull request merged**